fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -433,7 +434,13 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             splashScreen.setKeepOnScreenCondition {
                 val dataFetched = components.settings.utmParamsKnown &&
                     components.settings.nimbusExperimentsFetched
-                !maxDurationReached && !dataFetched
+                val keepOnScreen = !maxDurationReached && !dataFetched
+                if (!keepOnScreen) {
+                    SplashScreen.firstLaunchExtended.record(
+                        SplashScreen.FirstLaunchExtendedExtra(dataFetched = dataFetched),
```

Comment:
Should we be recording telemetry in a function that's passed to `setKeepOnScreenCondition`?  Are we sure that this is only going to be called once?

Reply:
It's a condition, so it shouldn't get called once we return `false`. From the doc:
`The splash will stay visible until the condition isn't met anymore.`

Is there a nimbus side way to ensure an even is fired once?

Reply:
I will take my guess that it's `lifetime: application` in the metrics file. Updated it

Reply:
It looks like `lifetime: application` adds some additional requirements to the event. removed